packages/react/src/state.tsx

```diff
@@ -0,0 +1,98 @@
+import { createContext, useReducer } from 'react'
+
+import { action } from './events'
+import { ActionDataState, JSONValue, StateDataState } from './template'
+
+export type LocalState = Record<string, JSONValue>
+
+type StateUpdate<T> = T | StateModifier
+type UpdateStateAction<T> = ActionDataState<['state-update', StateDataState<T>, StateUpdate<T>]>
+type StateModifier = {
```

Comment:
usually I would use a union type here `PayloadStateModifier | ToggleStateModifier` etc

Reply:
Sounds good. Let's change to union type.